common/djangoapps/entitlements/models.py

```diff
@@ -1,8 +1,122 @@
 import uuid as uuid_tools
+from datetime import datetime, timedelta
 
+import pytz
 from django.conf import settings
+from django.contrib.sites.models import Site
 from django.db import models
+
+from certificates.models import GeneratedCertificate  # pylint: disable=import-error
 from model_utils.models import TimeStampedModel
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class CourseEntitlementPolicy(models.Model):
+    """
+    Represents the Entitlement's policy for expiration, refunds, and regaining a used certificate
+    """
+
+    DEFAULT_EXPIRATION_PERIOD_DAYS = 450
+    DEFAULT_REFUND_PERIOD_DAYS = 60
+    DEFAULT_REGAIN_PERIOD_DAYS = 14
+
+    # Use a DurationField to calculate time as it returns a timedelta, useful in performing operations with datetimes
+    expiration_period = models.DurationField(
+        default=timedelta(days=DEFAULT_EXPIRATION_PERIOD_DAYS),
+        help_text="Duration in days from when an entitlement is created until when it is expired.",
+        null=False
+    )
+    refund_period = models.DurationField(
+        default=timedelta(days=DEFAULT_REFUND_PERIOD_DAYS),
+        help_text="Duration in days from when an entitlement is created until when it is no longer refundable",
+        null=False
+    )
+    regain_period = models.DurationField(
+        default=timedelta(days=DEFAULT_REGAIN_PERIOD_DAYS),
+        help_text=("Duration in days from when an entitlement is redeemed for a course run until "
+                   "it is no longer able to be regained by a user."),
+        null=False
+    )
+    site = models.ForeignKey(Site)
+
+    def get_days_until_expiration(self, entitlement):
+        """
+        Returns an integer of number of days until the entitlement expires.
+        Includes the logic for regaining an entitlement.
+        """
+        now = datetime.now(tz=pytz.UTC)
+        expiry_date = entitlement.created + self.expiration_period
+        days_until_expiry = (expiry_date - now).days
+        if not entitlement.enrollment_course_run:
+            return days_until_expiry
+        else:
+            course_overview = CourseOverview.get_from_id(entitlement.enrollment_course_run.course_id)
+            # Compute the days left for the regain
+            days_since_course_start = (now - course_overview.start).days
+            days_since_enrollment = (now - entitlement.enrollment_course_run.created).days
+
+            # We want to return whichever days value is less since it is then the more recent one
+            days_until_regain_ends = min(days_since_course_start, days_since_enrollment)
```

Comment:
Nit: maybe call this `days_since_regain_period_start`, since `days_until_regain_ends` sounds like `self.regain_period.days`.